[Term Entry] C# Math Functions .Sqrt() #1613 (#1623)
Codecademy/docs: content/c-sharp/concepts/math-functions/terms/sqrt/sqrt.md

---
Title: '.Sqrt()'
Description: 'Returns the square root of a number.'
Subjects:
    - 'Computer Science'
    - 'Code Foundations'
Tags:
    - 'Arithmetic'
    - 'Numbers'
    - 'Methods'   
CatalogContent:
    - 'learn-c-sharp'
    - 'paths/computer-science'
---

The **`Math.Sqrt()`** method returns the square root of the number given.

## Syntax

```pseudo

> public static double Sqrt(double x);

Reviewer: ```suggestion
Math.Sqrt(x);
```

```

The method takes only one parameter, the variable `x`, of type `double`. The function will return a positive value of type `double` unless the value passed is one of the following:
-If `x` is negative, the function will return `NaN` (not a number).
-If `x` is `NaN`, the function will return `NaN`.
-If `x` is `PositiveInfinity`, the function will return `PositiveInfinity`.

## Example

> The following example demonstrates the `Math.Sqrt()` method. Four different values are passed to method, and the return values are printed with the `Console.WriteLine()` method.

Reviewer: ```suggestion
The following example demonstrates the `Math.Sqrt()` method. Four different values are passed to the method, and the return values are printed with the `Console.WriteLine()` method.
```

> The following example will implement how `Math.Sqrt()` will take its given value in the function and return its square root by print it with the `Console.WriteLine()` method.

Reviewer: ```suggestion
The following example demonstrates the `Math.Sqrt()` method. Four different values are passed to method, and the return values are printed with the `Console.WriteLine()` method.
```

---
Title: '.Sqrt()'
Description: 'Returns the square root of a number.'
Subjects:
    - 'Computer Science'
    - 'Code Foundations'
Tags:

>     - 'Sqrt'

Reviewer: ```suggestion
    - 'Arithmetic'
    - 'Numbers'
    - 'Methods'   
```

CatalogContent:
    - 'learn-c-sharp'
    - 'paths/computer-science'
---

The **`Math.Sqrt()`** method can return the sqaure root of a specified number. If the value of the number happens to be equal to Nan, NegativeInfinity, or PositiveInfinity, that value is returned.

## Syntax

```pseudo
public static double Sqrt(double x);
```

> The `.Sqrt()` function will can take on one parameter at a time. As shown above _x_ is the value we want to find the sqaure root of.

Reviewer: ```suggestion
The method takes only one parameter, the variable `x`, of type `double`. The function will return a positive value of type `double` unless the value passed is one of the following:
-If `x` is negative, the function will return `NaN` (not a number).
-If `x` is `NaN`, the function will return `NaN`.
-If `x` is `PositiveInfinity`, the function will return `PositiveInfinity`.
```

## Example

The following example will implement how `Math.Sqrt()` will take its given value in the function and return its square root by print it with the `Console.WriteLine()` method.

```cs
using System;

> namespace MySqaureRoot

Reviewer: ```suggestion
namespace MySquareRoot
```

    {
        public class Example
        {
            public static void Main(string[] args)
            {
                double x = Math.Sqrt(144);

                Console.WriteLine(x);
            }    
        }
    }
```

> This example results in the following output:

Reviewer: I would suggest having only one example for this method. There isn't really any additional functionality that is being demonstrated by having both examples. I would say the second is more complete, so, you may want to use that one in place of the first and delete the codebyte.

Author: Thank you for the suggestions! I went ahead and deleted the codebyte and replaced the first example with the second one. Please let me know if there is anything else I need to add on here!
